icu4c/source/test/intltest/dtptngts.cpp

```diff
@@ -1557,4 +1558,67 @@ void IntlTestDateTimePatternGeneratorAPI::test_jConsistencyOddLocales() { // ICU
     }
 }
 
+void IntlTestDateTimePatternGeneratorAPI::testBestPattern() {
+    // generic test for DateTimePatternGenerator::getBestPattern() that can be used to test multiple
+    // bugs in the resource data
+    static const char* testCases[] = {
+        // ICU-21650: if the user asks for E, the minimum field length is 3, but if he asks for c or e, it's 1
```

Comment:
Why is so? couly you put a pointer to the UTS35 which mention such here or some text to explain your rational in the comment. It is not straightforward to conclude that from glazing the UTS35. It will be nice if you can put down your logic here so others can follow. Thanks

Reply:
See above.  I added a comment to the test that points to UTS 35.

Reply:
somehow I didn't see your comment. Did you forget to push?